crates/qasm3/src/lib.rs

```diff
@@ -150,11 +154,147 @@ pub fn load(
     loads(py, source, custom_gates, include_path)
 }
 
+#[derive(Debug, Clone)]
+struct DumpOptions {
+    includes: Vec<String>,
+    basis_gates: Vec<String>,
+    disable_constants: bool,
+    allow_aliasing: bool,
+    indent: String,
+}
+
+impl Default for DumpOptions {
+    fn default() -> Self {
+        Self {
+            includes: vec!["stdgates.inc".to_string()],
+            basis_gates: vec![],
+            disable_constants: true,
+            allow_aliasing: false,
+            indent: "  ".to_string(),
+        }
+    }
+}
+
+#[pyfunction]
+#[pyo3(signature = (circuit, /, kwargs=None))]
+pub fn dumps(
+    _py: Python,
+    circuit: &Bound<PyAny>,
+    kwargs: Option<&Bound<PyDict>>,
+) -> PyResult<String> {
```

Comment:
Making a fake `kwargs` dictionary like this is a bit unusual for a Rust interface.  You can specify keyword args properly using the `pyo3(signature = ...)` derive macro, like you have, including specifying defaults.  But here, since it's only called directly by Python space that already sets all the defaults (and only _intended_ to be called that way), I'm not sure that the `kwargs` indirection is achieving anything over just making all the configuration options explicitly required.